dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/visitor/generator/GenerateJspVisitor.java

```diff
@@ -275,59 +301,71 @@ protected void generateStaticSection() {
         writer.println("static {");
     }
 
-    protected void generateInitSection(ValidateJspResult validatorResult) 
+    protected void generateInitSection(ValidateJspResult validatorResult)
         throws JspCoreException {
         writer.println("}");
         writer.println();
         GeneratorUtils.generateInitSectionCode(writer, GeneratorUtils.JSP_FILE_TYPE, jspOptions);  //PM06063
-        writer.println();        
+        writer.println();
         GeneratorUtils.generateELFunctionCode(writer, validatorResult);
         writer.println();
     }
 
     protected void generateServiceInitSection(ValidateJspResult validatorResult, boolean genSessionVariable) {
         writer.println();
+
+        /*
+         *  Mark recommened using synchonized, but with a performance caveat
```

Comment:
What do you mean "Mark recommended", are we going to document this someplace? Do we have an issue opened if that is the case?

Reply:
It's all part of https://github.com/OpenLiberty/open-liberty/issues/20738.

isThreadSafe is deprecated, but deprecated means the behavior is planned for removal.  However, servlet 6.0 removed the SingleThreadModel.   So this behavior is removed after all.    So, by adding the synchronized keyword, it would try to keep the isThreadSafe behavior as before (but just without the SingleThreadModel interface).

Reply:
As Mark said, we're trying to achieve the same result without SingleThreadModel. See `https://github.com/jakartaee/pages/issues/206#issuecomment-934272204`